src/Sentry.Serilog/SentrySink.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using Sentry.Extensibility;
+using Sentry.Protocol;
+using Sentry.Reflection;
+using Serilog.Core;
+using Serilog.Events;
+
+namespace Sentry.Serilog
+{
+    public sealed class SentrySink : ILogEventSink, IDisposable
+    {
+        private readonly IFormatProvider _formatProvider;
+        private readonly Func<string, IDisposable> _initAction;
+        private volatile IDisposable _sdkHandle;
+
+        private readonly object _initSync = new object();
+
+        internal static readonly (string Name, string Version) NameAndVersion
+            = typeof(SentrySink).Assembly.GetNameAndVersion();
+
+        private static readonly string ProtocolPackageName = "nuget:" + NameAndVersion.Name;
+
+        internal IHub Hub { get; set; }
+
+        public string Dsn { get; set; }
+
+        public SentrySink(IFormatProvider formatProvider) : this(formatProvider, SentrySdk.Init, HubAdapter.Instance)
+        { }
+
+        internal SentrySink(
+            IFormatProvider formatProvider,
+            Func<string, IDisposable> initAction,
+            IHub hubGetter)
+        {
+            Debug.Assert(initAction != null);
+            Debug.Assert(hubGetter != null);
+
+            _formatProvider = formatProvider;
+            _initAction = initAction;
+            Hub = hubGetter;
+        }
+
+        public void Emit(LogEvent logEvent)
```

Comment:
Not entierly accurate. If you take a look here:

https://github.com/getsentry/sentry-dotnet/blob/6289801f352b52e859328607f36ac0940da6eed7/src/Sentry.Serilog/SentrySinkExtensions.cs#L12-L19

The `restrictedToMinimumLevel` option sets the minimum level for which `Emit` is called by Serilog. I'm not really sure what the best way would be to have a minimum breadcrumb level, since Serilog will not call `Emit` if the aforementioned option is, for example, set to Error and the miniumum breadcrumb level is set to Info.

We could override `restrictedToMinimumLevel` and set it to the lowest level and then add our own switch that checks it within `Emit`, but that seems to be a bit counterintuitive I think.

Reply:
`restrictedToMinimumLevel ` is the minimal level to either event or breadcrumb then.

For example: `restrictedToMinimumLevel ` = `Info`, we would capture breadcrumbs (for Info or higher) and send events (error or higher)

If `restrictedToMinimumLevel ` is Warning, we'd only capture breadcrumbs for Warnings or higher, and send events Error or higher.

Finally if `restrictedToMinimumLevel ` is Error, then breadcrumbs would be tracked only for Error or higher as well as sending events.

A logging integration with Sentry that sends all `Log..` to Sentry doesn't make much sense. Sentry is not a logging tool and flooding it with events that are not grouped together create a storm of noise (i.e: Notifications). 

Keeping the Info/Warn logs around (breadcrumbs) for a while to be sent in case there's an Error is, in my opinion, the best approach as it gives you the context you need to troubleshoot an issue.